pkg/binary/envoytest/util.go

```diff
@@ -15,79 +15,144 @@
 package envoytest
 
 import (
-	"context"
 	"fmt"
+	"math/rand"
+	"os"
 	"path/filepath"
+	"sync"
 	"syscall"
+	"testing"
 	"time"
 
-	"github.com/mholt/archiver"
+	"github.com/mholt/archiver/v3"
+	"github.com/stretchr/testify/require"
 	"github.com/tetratelabs/log"
 
 	"github.com/tetratelabs/getenvoy/pkg/binary"
 	"github.com/tetratelabs/getenvoy/pkg/binary/envoy"
 	"github.com/tetratelabs/getenvoy/pkg/manifest"
 )
 
-// Reference indicates the default Envoy version to be used for testing
-var Reference = "standard:1.11.0"
+// Reference indicates the default Envoy version to be used for testing.
+// 1.12.7 is the last version to match the Istio version we are using (see go.mod)
+var Reference = "standard:1.12.7"
+
+var once sync.Once
+var errorFetchingEnvoy error
+
+// FetchEnvoyAndRun retrieves the Envoy indicated by Reference only once. This is intended to be used with TestMain.
+// In CI, you can execute this to obviate latency during test runs: "go run cmd/getenvoy/main.go fetch standard:1.12.7"
+func FetchEnvoyAndRun(m *testing.M) {
+	once.Do(func() {
+		errorFetchingEnvoy = fetchEnvoy()
+	})
+
+	if errorFetchingEnvoy != nil {
+		fmt.Println(errorFetchingEnvoy)
+		os.Exit(1)
+	}
+
+	os.Exit(m.Run())
+}
+
+func fetchEnvoy() error {
+	key, err := manifest.NewKey(Reference)
+	if err != nil {
+		return fmt.Errorf("unable to make manifest key %v: %w", Reference, err)
+	}
+
+	r, err := envoy.NewRuntime()
+	if err != nil {
+		return fmt.Errorf("unable to make new envoy runtime: %w", err)
+	}
 
-// Fetch retrieves the Envoy indicated by Reference
-func Fetch() error {
-	key, _ := manifest.NewKey(Reference)
-	r, _ := envoy.NewRuntime()
 	if !r.AlreadyDownloaded(key) {
 		location, err := manifest.Locate(key)
 		if err != nil {
-			return fmt.Errorf("unable to retrieve manifest from %v: %v", manifest.GetURL(), err)
+			return fmt.Errorf("unable to retrieve manifest from %v: %w", manifest.GetURL(), err)
 		}
-		if err := r.Fetch(key, location); err != nil {
-			return fmt.Errorf("unable to retrieve binary from %v: %v", location, err)
+
+		err = r.Fetch(key, location)
+		if err != nil {
+			return fmt.Errorf("unable to retrieve binary from %v: %w", location, err)
 		}
 	}
 	return nil
 }
 
-// Run executes envoy and waits for it to be ready
-// It is blocking and will only return once ready (nil) or context timeout is exceeded (error)
-func Run(ctx context.Context, r binary.Runner, bootstrap string) error {
+// RunKillOptions allows customization of Envoy lifecycle.
+type RunKillOptions struct {
+	Bootstrap        string
+	ExpectedStatus   int
+	RetainDebugStore bool
+	SleepBeforeKill  time.Duration
+}
+
+// RequireRunKill executes envoy, waits for ready, sends sigint, waits for termination, then unarchives the debug directory.
+// It should be used when you just want to cycle through an Envoy lifecycle.
+func RequireRunKill(t *testing.T, r binary.Runner, options RunKillOptions) {
 	key, _ := manifest.NewKey(Reference)
```

Comment:
I would prefer to check error 🙂 
```suggestion
	key, err := manifest.NewKey(Reference)
	require.NoError(t, err)
```